rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,291 @@
+                sequence_index=target_log.log_index,
+                from_event_type=from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=self.eth,
+                amount=ZERO,
+                to_event_type=None,
+                to_event_subtype=None,
+                to_counterparty=CPT_BALANCER_V1,
+                to_notes=None,
+                extra_data={
+                    'ds_address': ds_address,
+                    'token_address': token_address,
+                    'amount': amount,
+                    'type': balancer_event_type,
+                },
+            )
+            action_items.append(action_item)
+
+        return action_items
+
+    def _decode_swap_creation(
+            self,
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+            action_items: list[ActionItem],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        # The transfer event appears after the debt generation event, so we need to transform it
+        from_token = hex_or_bytes_to_address(tx_log.topics[2])
+        to_token = hex_or_bytes_to_address(tx_log.topics[3])
+        amount_in = hex_or_bytes_to_int(tx_log.data[0:32])
+        amount_out = hex_or_bytes_to_int(tx_log.data[32:64])
+        action_item = ActionItem(
+            action='transform',
+            sequence_index=tx_log.log_index,
+            from_event_type=HistoryEventType.RECEIVE,
+            from_event_subtype=HistoryEventSubType.NONE,
+            asset=self.eth,
+            amount=ZERO,
+            to_event_type=HistoryEventType.WITHDRAWAL,
+            to_event_subtype=HistoryEventSubType.GENERATE_DEBT,
+            to_counterparty=CPT_BALANCER_V2,
+            to_notes=None,
+            extra_data={
+                'from_token': from_token,
+                'to_token': to_token,
+                'amount_in': amount_in,
+                'amount_out': amount_out,
+            },
+        )
+
+        amount_in_if_eth = asset_normalized_value(
+            amount=amount_in,
+            asset=self.eth,
+        )
+
+        if len(action_items) == 0 and from_token == self.weth.evm_address:
+            for event in decoded_events:
+                if (
+                    event.asset == A_ETH and event.balance.amount == amount_in_if_eth and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.TRADE
+                    event.event_subtype = HistoryEventSubType.SPEND
+                    event.notes = f'Swap {event.balance.amount} {self.eth.symbol} in balancer V2 from {event.location_label}'  # noqa: E501
+                    event.counterparty = CPT_BALANCER_V2
+
+        return None, [action_item]
+
+    def decode_balancer_v2_event(
```

Comment:
This function seems superficial. Just add the negated condition on the top of `_decode_swap_creation` with a `return None, []` and you can use only that and save space.